lambdas/service/app.py

```diff
@@ -655,204 +661,237 @@ def repository_search(entity_type: str, item_id: str):
+                    for relation in ('is', 'contains', 'within', 'intersects')
+                })
+            ) for facet in facets
+        }),
+        description=format_description('''
+            Filters to be applied when calling ElasticSearch. Omit
+            parameter or use `{}` to skip filtering. The relationship
+            parameters are always nested within a singleton array, so
+            use `{"is":["foo"]}` to check for equality with "foo" and
+            `{"within":[[0, 100]]}` to check for falling within the
+            range [0, 100].
+        '''))
+
+
+def repository_search_spec(entity_type):
+    id_spec_link = f'#operations-Repository-get_repository_{entity_type}s__{entity_type}_id_'
+    facets = sorted(valid_facets())
+    return {
+        'summary': f'Search the {entity_type}s index for entities of interest.',
+        'tags': ['Repository'],
+        'parameters': [
+            repository_filters_spec(facets),
+            params.query(
+                'size',
+                schema.optional(int),
+                description='The number of hits included per page.'),
+            params.query(
+                'sort',
+                schema.optional(schema.enum(*facets)),
+                description='The facet to sort the hits by.'),
+            params.query(
+                'order',
+                schema.optional(schema.enum('asc', 'desc')),
+                description=format_description('''
+                            The order of the hits when sorting, either ascending
+                            or descending. Has no effect unless `sort` is provided.
+                ''')),
+            *[
+                params.query(
+                    param,
+                    schema.optional(str),
+                    description=format_description('''
+                        Use the `next` and `previous` properties of the
+                        `pagination` response element to navigate between pages.
+                    '''),
+                    deprecated=True)
+                for param in ['search_before', 'search_before_uid', 'search_after', 'search_after_uid']
+            ]
+        ],
+        'responses': {
+            '200': {
+                'description': format_description(f'''
+                    Paginated list of {entity_type}s that meet the search
+                    criteria ("hits"). The structure of these hits is documented
+                    under the [corresponding endpoint for a specific entity]({id_spec_link}).
+
+                    The `pagination` section describes the total number of hits
+                    and total number of pages, as well as user-supplied search
+                    parameters for page size and sorting behavior. It also
+                    provides links for navigating forwards and backwards between
+                    pages of results.
+
+                    The `termFacets` section tabulates the occurrence of unique
```

Comment:
Somehow it needs to be documented that termFacets doesn't just cover the current page, but all entities matching the filter.